### PR TITLE
Security policy and process for the project

### PR DESCRIPTION
**Pull request by toddysm**
@vaninrao10 and I have been collaborating on creating the Security Policy for the project. This text is the draft that we ended up with and we are submitting it for review to the project governance maintainers.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.vscode/*
```
Comment (yizha1) [inline .gitignore:1]: Is this file needed?

Comment (yizha1) [inline .gitignore:1]: Is this file needed?

File: SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+# Notation Security Process and Policy
+This document provides the details on the Notation security policy and details the process surrounding security handling including a how to report a security vulnerability for anything within the Notary Project organization.
```
Comment (yizha1) [inline SECURITY.md:2]: nit:
extra `a` before `how to`

Comment (yizha1) [inline SECURITY.md:2]: Seems only Notation related repo were listed below for VA report, should we mention `Notary Project organization`? Will `notary` repo be covered?

File: SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+- Members SHOULD engage in each reported vulnerability, at a minimum to make sure it is being handled
+- Members MUST keep the vulnerability details private and only share on a need to know basis
+
+### Membership
+New members are required to be active maintainers of Notation projects who are willing to perform the responsibilities outlined above. The security team is a subset of the maintainers. Members can step down at any time and may join at any time.
+
+From time to time, Notation projects are deprecated. If at any time a security team member is found to be no longer be an active maintainer on active Notation projects, this individual will be removed from the security team.
+
+## Patch and Release Team
+When a vulnerability comes in and is acknowledged, a team - including maintainers of the Notation project affected - will be assembled to patch the vulnerability, release an update, and publish the vulnerability disclosure. This may expand beyond the security team as needed but will stay within the pool of Notation project maintainers.
```
Comment (yizha1) [inline SECURITY.md:90]: Here `Notation project` is mentioned, not `notary project` at the beginning of this doc. It's better to be consistent. Since we only cover `Notation` repos, my suggestion is to use `Notation project`.

File: SECURITY.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+Here are the links to security advisories for each repository:
+
+- [Notation CLI Security Advisories](https://github.com/notaryproject/notation/security/advisories)
+- [Notation-Go Library Security Advisories](https://github.com/notaryproject/notation-go/security/advisories)
+- [Notation-Core-Go Library Security Advisories](https://github.com/notaryproject/notation-go/security/advisories)
+- [Notation Spcecification Security Advisories](https://github.com/notaryproject/notaryproject/security/advisories)
+
+## Security Team Membership
+The security team is made up of a subset of the Notation project maintainers who are willing and able to respond to vulnerability reports.
```
Comment (yizha1) [inline SECURITY.md:77]: Are the sub-repo maintainers by default the members of the security team? Or do we need to nominate people to security team per sub-repo?

**Review by yizha1 — COMMENTED** [review]
Thanks @toddysm. Some comments provided.

File: .gitignore (hunk 1 above)
Comment (toddysm) [reply .gitignore:1]: `.gitignore` is good to have so we don't pollute the repository with non-relevant files.

File: SECURITY.md (hunk 2 above)
Comment (toddysm) [reply SECURITY.md:2]: I believe this is grammatically correct - the `a` relates to the `report` noun.

File: SECURITY.md (hunk 3 above)
Comment (toddysm) [reply SECURITY.md:90]: This Security Policy will apply to all sub-projects under Notary Project, including Notation. I fixed all references.

File: SECURITY.md (hunk 4 above)
Comment (toddysm) [reply SECURITY.md:77]: The sub-project maintainers are contacted by default. Subset can acknowledge and add additional members if relevant.

Comment (toddysm) [reply SECURITY.md:77]: Updated the text to clarify.

File: SECURITY.md (hunk 2 above)
Comment (toddysm) [reply SECURITY.md]: This is a good point. I added the other two repos that have code in them.

**Review by yizha1 — APPROVED** [review]
LGTM

**Review by vaninrao10 — DISMISSED** [review]
LGTM

**Comment by iamsamirzon** [comment]
LGTM but IANAM

File: .gitignore (hunk 1 above)
Comment (priteshbandi) [reply .gitignore:1]: IMO the  committed `.gitignore` file should include names of files/directories that are created when working on the code directly and should not be committed.

It should not include names of files that are only created by personal tools such as vscode, GoLand, intelliJ etc. For these, we can use a system-wide [Global gitignore](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer)

**Review by priteshbandi — DISMISSED** [review]
LGTM

**Comment by anvega** [comment]
From chatting with Toddy, I pointed out it's a good idea with the interest of clarity to lay out the names of maintainers who compose the security team in a table. It's common for projects to do that so reporters have a sense of who the private reports are going to. The way the text currently reads says it's a subset although the way that GitHub disclosure reports are set up, it notifies everyone with a commit bit. In addition and or as an alternative, you could also define the subteam in the maintainers.md file.

**Review by FeynmanZhou — DISMISSED** [review]
LGTM

**Review by anvega — APPROVED** [review]
This reads much better. Thanks for clarifying.

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by vaninrao10 — APPROVED** [review]
LGTM

**Review by NiazFK — APPROVED** [review]
LGTM

**Review by yizha1 — APPROVED** [review]
LGTM but IANAM

**Review by FeynmanZhou — APPROVED** [review]
LGTM
